engine/api/pipeline/pipeline_build.go

```diff
@@ -553,41 +520,40 @@ func InsertPipelineBuild(tx gorp.SqlExecutor, project *sdk.Project, p *sdk.Pipel
 
 		if !found {
 			//If git.branch was not found is pipeline parameters, we set de previously found defaultBranch
-			params = append(params, sdk.Parameter{
-				Name:  "git.branch",
-				Value: defautlBranch,
-				Type:  sdk.StringParameter,
-			})
+			sdk.AddParameter(&params, "git.branch", sdk.StringParameter, defautlBranch)
 			pb.Trigger.VCSChangesBranch = defautlBranch
 
 			//And we try to put the lastestCommit for this branch
 			if lastGitHash[defautlBranch] != "" {
-				params = append(params, sdk.Parameter{
-					Name:  "git.hash",
-					Value: lastGitHash[defautlBranch],
-					Type:  sdk.StringParameter,
-				})
+				sdk.AddParameter(&params, "git.hash", sdk.StringParameter, lastGitHash[defautlBranch])
 				pb.Trigger.VCSChangesHash = lastGitHash[defautlBranch]
 			}
 		} else {
 			//If git.branch was found but git.hash wasn't found in pipeline parameters
 			//we try to found the LatestCommit
 			if !hashFound && lastGitHash[pb.Trigger.VCSChangesBranch] != "" {
-				params = append(params, sdk.Parameter{
-					Name:  "git.hash",
-					Value: lastGitHash[pb.Trigger.VCSChangesBranch],
-					Type:  sdk.StringParameter,
-				})
+				sdk.AddParameter(&params, "git.hash", sdk.StringParameter, lastGitHash[pb.Trigger.VCSChangesBranch])
 				pb.Trigger.VCSChangesHash = lastGitHash[pb.Trigger.VCSChangesBranch]
 			}
 		}
 	}
 
+	//Retreive commit information
+	if client != nil && pb.Trigger.VCSChangesHash != "" {
+		commit, err := client.Commit(app.RepositoryFullname, pb.Trigger.VCSChangesHash)
+		if err != nil {
+			log.Warning("InsertPipelineBuild> Cannot get commit: %s\n", err)
+			return nil, err
+		}
+		sdk.AddParameter(&params, "git.author", sdk.StringParameter, commit.Author.Name)
```

Comment:
but it's triggered on only one commit identified by its hash, even if it can embed more than commit.
as you can see, git.author represents the author of the git.hash